reinforcement_learning/rlclientlib/logger/event_batcher.h

```diff
@@ -0,0 +1,87 @@
+#include "event_queue.h"
+#include "ranking_event.h"
+#include "generated/RankingEvent_generated.h"
+#include "generated/OutcomeEvent_generated.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+    template<typename TEvent>
+    class event_batcher {
+    public:
+      event_batcher() {};
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size);
+    };
+
+    template<typename TEvent>
+    void event_batcher<TEvent>::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size) {
```

Comment:
it is not clear how responsibilities of event_batcher and async_batcher are related to each other? With current implementation event_batcher cannot be used out of existing scenario because it is tightly coupled with async_batcher implementation details (event_queue / send_high_water_mark/etc). In this case we probably do not need separate class and just additional private method inside async_batcher is ok. 
Other option is to make them orthogonal to each other - but in this case only event_batcher should be owner of all batch-related parameters (send_high_watermark / batch_size / ...)

Reply:
I agree and I think I would move them into async_batcher. The reason why we had that was we might want to support both ds-json string and flatbuffer at the same time. But this was not the case anymore. 

---
In reply to: [228219231](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1653#discussion_r228219231) [](ancestors = 228219231)